diff --git a/src/video/cocoa/SDL_cocoawindow.m b/src/video/cocoa/SDL_cocoawindow.m
index 41a27fd4..1dbf3023 100644
--- a/src/video/cocoa/SDL_cocoawindow.m
+++ b/src/video/cocoa/SDL_cocoawindow.m
@@ -1054,19 +1054,22 @@ static NSCursor *Cocoa_GetDesiredCursor(void)
 {
     SDL_Window *window = _data.window;
     SDL_Surface *shape = (SDL_Surface *)SDL_GetPointerProperty(SDL_GetWindowProperties(window), SDL_PROP_WINDOW_SHAPE_POINTER, NULL);
-    BOOL ignoresMouseEvents = NO;
 
-    if (shape) {
-        NSPoint point = [theEvent locationInWindow];
-        NSRect windowRect = [[_data.nswindow contentView] frame];
-        if (NSMouseInRect(point, windowRect, NO)) {
-            int x = (int)SDL_roundf((point.x / (window->w - 1)) * (shape->w - 1));
-            int y = (int)SDL_roundf(((window->h - point.y) / (window->h - 1)) * (shape->h - 1));
-            Uint8 a;
+    if (!shape) {
+        return;
+    }
 
-            if (!SDL_ReadSurfacePixel(shape, x, y, NULL, NULL, NULL, &a) || a == SDL_ALPHA_TRANSPARENT) {
-                ignoresMouseEvents = YES;
-            }
+    BOOL ignoresMouseEvents = NO;
+    
+    NSPoint point = [theEvent locationInWindow];
+    NSRect windowRect = [[_data.nswindow contentView] frame];
+    if (NSMouseInRect(point, windowRect, NO)) {
+        int x = (int)SDL_roundf((point.x / (window->w - 1)) * (shape->w - 1));
+        int y = (int)SDL_roundf(((window->h - point.y) / (window->h - 1)) * (shape->h - 1));
+        Uint8 a;
+
+        if (!SDL_ReadSurfacePixel(shape, x, y, NULL, NULL, NULL, &a) || a == SDL_ALPHA_TRANSPARENT) {
+            ignoresMouseEvents = YES;
         }
     }
     _data.nswindow.ignoresMouseEvents = ignoresMouseEvents;
